src/charm.py

```diff
@@ -83,8 +88,20 @@ def __init__(self, *args):
         self.redis = RedisRequires(self, self._stored)
         self.framework.observe(self.on.redis_relation_updated, self._config_changed)
 
-    def _make_ingress_config(self):
-        """Return a dict of our ingress config."""
+        self._metrics_endpoint = MetricsEndpointProvider(
+            self, jobs=[{"static_configs": [{"targets": ["*:9394"]}]}]
```

Comment:
This port sounds like a magic number, is there a way to specify the port to the exporter service and therefore set the port to a constant value to ensure it'll always match ?

Reply:
That would mean some additional complexity. Passing that value to the script and the to the appropriate command, instead I decided to rely on the default configuration.

Reply:
We have several unpinned dependencies and I'd say an issue like that is as unlikely to happen as any other dependency  change breaking discourse. I'd rather be consistent here

Reply:
Well still not convinced as it's still a magic number that could lead to a break of the feature for no apparent reason by building a new docker image (which doesn't go through the integration test pipeline).
But if you're sure then so be it

Just one question, how does the charm behave if nothing is listening on the configured port ?

Reply:
It will try to scrape that port anyway. We have an alert that will be triggered if it fails to do so.

Reply:
I've added an integration test that should catch this, btw

Reply:
Closing this discussion, at least there's a feedback for the user when it can't connect.
The integration test is good, too thx